net/nimble/controller/src/ble_ll_test.c

```diff
@@ -0,0 +1,498 @@
+                0xab, 0x02, 0xa5, 0x9e, 0xee, 0xc0, 0xb9, 0xe4,
+                0x75, 0x45, 0xc2, 0x1c, 0x7a, 0x9b, 0x85, 0x7b,
+                0xf8, 0x0f, 0xde, 0x47, 0x67, 0x82, 0x94, 0x5b,
+                0x3c, 0x9f, 0x8d, 0x4a, 0x1c, 0x5b, 0x1d, 0x19,
+                0x11, 0x20, 0xc4, 0x90, 0x53, 0x0d, 0x7b, 0xd9,
+                0x89, 0x42, 0x2d, 0xbf, 0x49, 0xda, 0x4f, 0x56,
+                0x66, 0xc0, 0x98, 0x62, 0xe9, 0x2f, 0x1a, 0xd7,
+                0x34, 0x8f, 0xef, 0x82, 0xb5, 0xdd, 0xa0, 0xf5,
+                0x55, 0x81, 0x52, 0x4f, 0x77, 0xe0, 0x5c, 0xf2,
+                0xba, 0x22, 0x61, 0x0e, 0xbd, 0xcd, 0xc2 };
+
+#define BLE_TEST_SYNC_WORD          (0x71764129)
+#define BLE_TEST_CRC                (0x555555)
+
+static void
+ble_ll_test_set_next(struct test_ctx *ctx)
+{
+    struct ble_ll_sched_item *sch = &ctx->sch;
+
+    sch->start_time += ctx->itvl_ticks;
+    sch->remainder += ctx->itvl_rem_usec;
+    if (sch->remainder > 30) {
+       sch->start_time++;
+       sch->remainder -= 30;
+    }
+
+    sch->start_time -= g_ble_ll_sched_offset_ticks;
+}
+
+static void
+ble_ll_test_event(struct os_event *evt) {
+    /* It is called in LL context */
+    struct test_ctx *c = evt->ev_arg;
+    int rc;
+    os_sr_t sr;
+
+    OS_ENTER_CRITICAL(sr);
+    if (!c->active) {
+        OS_EXIT_CRITICAL(sr);
+        return;
+    }
+
+    if (!c->om) {
+        OS_EXIT_CRITICAL(sr);
+        return;
+    }
+
+    ble_ll_test_set_next(c);
+    rc = ble_ll_sched_dtm(&c->sch);
+    assert(rc == 0);
+    OS_EXIT_CRITICAL(sr);
+}
+
+static void
+ble_ll_tx_done(void *arg)
+{
+    struct test_ctx *c;
+
+    c = arg;
+    if (!c->evt.ev_cb) {
+        return;
+    }
+    c->num_of_packets++;
+    /* Reschedule event in LL context */
+    os_eventq_put(&g_ble_ll_data.ll_evq, &c->evt);
+
+    ble_ll_state_set(BLE_LL_STATE_STANDBY);
+}
+
+static int
+ble_ll_test_tx_sched_cb(struct ble_ll_sched_item *sch)
+{
+    struct test_ctx *ctx = sch->cb_arg;
+    int rc;
+
+    if (!ctx->active) {
+        return BLE_LL_SCHED_STATE_DONE;
+    }
+
+    rc = ble_phy_setchan(ctx->chan, BLE_TEST_SYNC_WORD, BLE_TEST_CRC);
+    if (rc != 0) {
+        assert(0);
+        return BLE_LL_SCHED_STATE_DONE;
+    }
+
+    ble_phy_mode_set(ctx->phy_mode, ctx->phy_mode);
+    ble_phy_set_txend_cb(ble_ll_tx_done, ctx);
+    ble_phy_txpwr_set(0);
+
+    sch->start_time += g_ble_ll_sched_offset_ticks;
+
+    /*XXX Maybe reschedule if too late */
+    rc = ble_phy_tx_set_start_time(sch->start_time, sch->remainder);
+    assert(rc == 0);
+
+    rc = ble_phy_tx(ctx->om, BLE_PHY_TRANSITION_NONE);
+    assert(rc == 0);
+
+    ble_ll_state_set(BLE_LL_STATE_DTM);
+
+    return BLE_LL_SCHED_STATE_DONE;
+}
+
+static void
+ble_ll_test_calculate_itvl(struct test_ctx *ctx, uint8_t len, int phy_mode)
+{
+    uint32_t l;
+    uint32_t itvl_usec;
+    uint32_t itvl_ticks;
+
+    /* Calculate interval */
+    l = ble_ll_pdu_tx_time_get(len + BLE_LL_PDU_HDR_LEN, phy_mode);
```

Comment:
some more verbose variable name would be nice

Reply:
I would keep it L as it is in spec. Added comment where you can find it in spec.